### PR TITLE
Authorizer refactor preparation (tests)

### LINKED ISSUE #1964: Remove Authorizer(Adaptor) deployments from tests
Pending changes to the Authorizer would break tests that rely on external deployments of the Authorizer/AuthorizerAdaptor, so make all tests read these contracts from the Vault deployment.

### PR DESCRIPTION
**Pull request by EndymionJkb**
# Description

In preparation for wider changes to the veBAL system, remove explicit deployments of the Authorizer and AuthorizerAdaptor from tests. The Vault(.ts) will now deploy these, and contracts that need them can ask this object vs having the test deploy and pass them in, as many of the tests were doing.

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [X] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [X] The diff is legible and has no extraneous changes
- [X] Complex code has been commented, including external interfaces
- [X] Tests are included for all code paths
- [X] The base branch is either `master`, or there's a description of how to merge

## Issue Resolution

Closes #1964

### CODE REVIEW

File: pkg/pool-utils/test/BasePool.test.ts [hunk 1]
```diff
@@ -32,8 +33,9 @@ describe('BasePool', function () {
   });
 
   sharedBeforeEach(async () => {
-    authorizer = await deploy('v2-vault/TimelockAuthorizer', { args: [admin.address, ZERO_ADDRESS, MONTH] });
-    vault = await deploy('v2-vault/Vault', { args: [authorizer.address, ZERO_ADDRESS, 0, 0] });
+    const vaultObj = await Vault.create({ admin });
+    vault = vaultObj.instance;
+    authorizer = vaultObj.authorizer;
```
Comment (nventuro) [inline pkg/pool-utils/test/BasePool.test.ts:38]:
```suggestion
    ({ instance: vault, authorizer } = await Vault.create({ admin }));
```

Comment (nventuro) [reply pkg/pool-utils/test/BasePool.test.ts:38]: You can use this form to reduce verbosity and boilerplate a notch.

File: pkg/governance-scripts/test/BaseCoordinator.test.ts [hunk 2]
```diff
@@ -25,8 +23,9 @@ describe('BaseCoordinator', () => {
   sharedBeforeEach('deploy vault', async () => {
     vault = await Vault.create({ admin });
     if (!vault.authorizer) throw Error('Vault has no Authorizer');
+    if (!vault.authorizerAdaptor) throw Error('Vault has no AuthorizerAdaptor');
```
Comment (nventuro) [inline pkg/governance-scripts/test/BaseCoordinator.test.ts:26]: Do we need this? It seems these values are not optional - we always have them. If so you can use the alternate format suggested in the other comment to condense all of this to a single line.

**Review by nventuro — APPROVED** [review]

File: pkg/governance-scripts/test/BaseCoordinator.test.ts (hunk 2 above)
Comment (EndymionJkb) [reply pkg/governance-scripts/test/BaseCoordinator.test.ts:26]: A few of them have these, so I just followed the pattern. (Authorizer is also not optional.) Happy to remove them.

**Pull request merged**